src/App.tsx

```diff
@@ -17,18 +43,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                queryValue={query}
+                filterValue={status}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todosRedux.length ? (
+                <TodoList
+                  todos={visibleTodos}
+                  selectedTodoId={currentTodo?.id || 0}
+                />
+              ) : (
+                <Loader />
+              )}
```

Comment:
if server will return you empty array - loader always will be visible.

use isLoading state